qltui.py

```diff
@@ -0,0 +1,743 @@
+        buff = ql.mem.write(user_data["address"], user_data["value"])
+        ql.log.info(f"Hook was triggered at -> {user_data['address']}")
+        ql.log.info(buff)
+
+    @staticmethod
+    def write_reg_2_args(ql,user_data):
+        buff = ql.reg.write(user_data["register_name"], user_data["value"])
+        ql.log.info(f"Hook was triggered at -> {user_data['register_name']}")
+        ql.log.info(buff)
+
+    @staticmethod
+    def write_reg_4_args(ql, address=0, size=0, user_data=None):
+        buff = ql.reg.write(user_data["register_name"], user_data["value"])
+        ql.log.info(f"Hook was triggered at -> {user_data['register_name']}")
+        ql.log.info(buff)
+
+    @staticmethod
+    def write_reg_6_args(ql, address=0, size=0, access=1, value=0, user_data=None):
+        buff = ql.reg.write(user_data["register_name"], user_data["value"])
+        ql.log.info(f"Hook was triggered at -> {user_data['register_name']}")
+        ql.log.info(buff)
+
+    @staticmethod
+    def emu_start_2_args(ql, user_data):
+        ql.emu_start(begin=user_data["start"], end=user_data["end"])
+
+    @staticmethod
+    def emu_start_4_args(ql, address=0, size=0, user_data=None):
+        ql.emu_start(begin=user_data["start"], end=user_data["end"])
+
+    @staticmethod
+    def emu_start_6_args(ql, address=0, size=0, access=1, value=0, user_data=None):
+        ql.emu_start(begin=user_data["start"], end=user_data["end"])
+
+    @staticmethod
+    def emu_stop_2_args(ql, user_data=None):
+        ql.log.info('killer switch found, stopping')
+        ql.emu_stop()
+
+    @staticmethod
+    def emu_stop_4_args(ql, address=0, size=0, user_data=None):
+        ql.log.info('killer switch found, stopping')
+        ql.emu_stop()
+
+    @staticmethod
+    def emu_stop_6_args(ql, address=0, size=0, access=1, value=0, user_data=None):
+        ql.log.info('killer switch found, stopping')
+        ql.emu_stop()
+
+    @staticmethod
+    def save_2_args(ql, user_data=None):
+        ql.save()
+
+    @staticmethod
+    def save_4_args(ql, address=0, size=0, user_data=None):
+        ql.save()
+
+    @staticmethod
+    def save_6_args(ql, address=0, size=0, access=1, value=0, user_data=None):
+        ql.save()
+
+def env_arg(value):
+    """
+    Function to read env parameter
+    """
+    if value == "{}":
+        return {}
+    else:
+        if os.path.exists(value):
+            with open(value, 'rb') as f:
+                env = pickle.load(f)
+        else:
+            env = ast.literal_eval(value)
+        return env
+
+
+def verbose_arg(value):
+    """
+    Function to map Verbose
+    """
+    return verbose_map[value]
+
+
+def read_file(fname: str):
+    """
+    Function to read code from file
+    """
+    with open(fname, "rb") as f:
+        content = f.read()
+
+    return content
+
+
+def log(string, color):
+    """
+    Function to beautify terminal output
+    """
+    if colored:
```

Comment:
`colored` is constant throughout the emulation. Instead of testing for it every time a log line is about to be emitted, it could be determined once and choose between two methods to implement it (i.e. the `if` clause or the `else` one).